zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/internal/BulkIndexWriter.java

```diff
@@ -73,103 +79,113 @@
-      writer.name("id").value(span.id());
-      if (span.kind() != null) writer.name("kind").value(span.kind().toString());
-      if (span.name() != null) writer.name("name").value(span.name());
-      if (span.timestampAsLong() != 0L) writer.name("timestamp").value(span.timestampAsLong());
-      if (span.durationAsLong() != 0L) writer.name("duration").value(span.durationAsLong());
+      writer.writeStringField("traceId", span.traceId());
+      if (span.parentId() != null) writer.writeStringField("parentId", span.parentId());
+      writer.writeStringField("id", span.id());
+      if (span.kind() != null) writer.writeStringField("kind", span.kind().toString());
+      if (span.name() != null) writer.writeStringField("name", span.name());
+      if (span.timestampAsLong() != 0L) {
+        writer.writeNumberField("timestamp", span.timestampAsLong());
+      }
+      if (span.durationAsLong() != 0L) writer.writeNumberField("duration", span.durationAsLong());
       if (span.localEndpoint() != null && !EMPTY_ENDPOINT.equals(span.localEndpoint())) {
-        writer.name("localEndpoint");
+        writer.writeFieldName("localEndpoint");
         write(span.localEndpoint(), writer);
       }
       if (span.remoteEndpoint() != null && !EMPTY_ENDPOINT.equals(span.remoteEndpoint())) {
-        writer.name("remoteEndpoint");
+        writer.writeFieldName("remoteEndpoint");
         write(span.remoteEndpoint(), writer);
       }
       if (!span.annotations().isEmpty()) {
-        writer.name("annotations");
-        writer.beginArray();
+        writer.writeArrayFieldStart("annotations");
         for (int i = 0, length = span.annotations().size(); i < length; ) {
           write(span.annotations().get(i++), writer);
         }
-        writer.endArray();
+        writer.writeEndArray();
       }
       if (!span.tags().isEmpty()) {
-        writer.name("tags");
-        writer.beginObject();
+        writer.writeObjectFieldStart("tags");
         Iterator<Map.Entry<String, String>> tags = span.tags().entrySet().iterator();
         while (tags.hasNext()) write(tags.next(), writer);
-        writer.endObject();
+        writer.writeEndObject();
       }
-      if (Boolean.TRUE.equals(span.debug())) writer.name("debug").value(true);
-      if (Boolean.TRUE.equals(span.shared())) writer.name("shared").value(true);
-      writer.endObject();
-      writer.flush();
-      hashingSink.flush();
+      if (Boolean.TRUE.equals(span.debug())) writer.writeBooleanField("debug", true);
+      if (Boolean.TRUE.equals(span.shared())) writer.writeBooleanField("shared", true);
+      writer.writeEndObject();
     } catch (IOException e) {
       throw new AssertionError(e); // No I/O writing to a Buffer.
     }
-    return new Buffer()
-      .writeUtf8(span.traceId()).writeByte('-').writeUtf8(hashingSink.hash().hex())
-      .readUtf8();
+
+    ByteBuf slice = sink.buffer().slice(startIndex, sink.buffer().writerIndex() - startIndex);
```

Comment:
```suggestion
    // get a slice representing the document we just wrote so that we can make a content hash
    ByteBuf slice = sink.buffer().slice(startIndex, sink.buffer().writerIndex() - startIndex);
```